src/plugins/visualize/public/application/editor/editor.js

```diff
@@ -253,6 +253,35 @@ function VisualizeAppController($scope, $route, $injector, $timeout, kbnUrlState
                 });
               };
 
+              const createVisReference = () => {
+                const currentDashboardInput = dashboard.getLastLoadedDashboardAppDashboardInput();
```

Comment:
i think we should not be loading the dashboard and manipulating it in visualize (this requires internal knowledge about dashboard structure)

rather we should navigate to dashboard and let dashboard take care of that:

`navigateToApp('dashboard', { newVis: vis.serialize(); })`

and then dashboard should check for `newVis` on its state, if its there it can append it to its panels

Reply:
I agree. The problem here is that we started working on a better inter-app communication logic and the PR hasn't been merged yet. As I mentioned in the description, this code is definitely going to change after https://github.com/elastic/kibana/pull/67064 is merged and what you're describing is pretty similar to how it is going to work. We'll navigate to dashboard and pass the serialized vis as input, and the dashboard is going to render.

I did not want to base this PR off an unmerged PR, but I also didn't want to be blocked by that change. 

I think this way of communicating between dashboard and visualization is good enough for now.